### DISCUSSION TITLE
Get bean validation to work

### DISCUSSION THREAD

**Discussion by gabriel-hoff-vectorsolutions** [Netflix/dgs-framework#1425] [Q&A]
Is it possible to use `@Valid` or `@Validated` annotations in mutations? For example:

```java
public class ProductInput {

    @NotBlank
    @Size(max = 50)
    private String name;

}
```

```java
@DgsMutation
public Product createProduct(@InputArgument @Valid ProductInput input) {
    return productService.create(input);
}
```

**A bit of context:**

I understand that DGS supports [graphql-java-extended-validation](https://github.com/graphql-java/graphql-java-extended-validation).
Still, my team will be migrating existing REST APIs to GraphQL, and everything is being validated using bean validation (via Hibernate validator). I'm trying to assess the complexity of the migration, and not having to switch to something like `graphql-java-extended-validation` from the get-go would probably help a lot.

**Edit:**
Based on existing discussions, it seems that used to work at some level (unless these people were getting false positives):
- https://github.com/Netflix/dgs-framework/discussions/620
- https://github.com/Netflix/dgs-framework/discussions/379

**Comment by torbenkeller** [comment]
For me it worked when I annotated my Controller with `@Validated` and the input argument with `@Valid`
